libvast/src/system/indexer.cpp

```diff
@@ -106,6 +130,17 @@ indexer(caf::stateful_actor<indexer_state>* self, path filename,
     [=](const std::vector<table_slice_column>& xs) {
       handle_batch(xs); // clang-format fix
     },
+    [=](telemetry_atom) {
+      self->state.send_report();
+      // The indexers are relying on caf's reference counting to
+      // shut down, so we stop telemetry sending when the partition
+      // is removed as it will keep a reference to this indexer alive;
+      // thus the actor will never quit and block the shutdown of caf.
+      if (!self->state.stop_telemetry)
+        self->delayed_send(self, defaults::system::telemetry_rate,
+                           telemetry_atom::value);
+    },
+    [=](stop_atom) { self->state.stop_telemetry = true; },
     [=](shutdown_atom) {
       self->quit(caf::exit_reason::user_shutdown); // clang-format fix
     },
```

Comment:
I noticed that `shutdown_atom` is currently unused, not sure whether we should remove it completely or keep it just in case.